main.cpp

```diff
@@ -55,39 +56,47 @@ GLFWwindow* SetupGLFW()
     GLFWwindow* window = glfwCreateWindow(640, 480, "Hello World", nullptr, nullptr);
     if (window == nullptr)
     {
-        std::cerr << "glfwCreateWindow() failure" << std::endl;
+        spdlog::error("glfwCreateWindow() failure");
         glfwTerminate();
         return nullptr;
     }
 
     glfwMakeContextCurrent(window);
     glfwSetKeyCallback(window, Input::KeyCallback);
 
-    //Uncomment this if you would like to blow up the GPU :D
-    // glfwSwapInterval(0);
+    bool fpsCapped = true;
+    if (jsonData.find("fps_capped") != jsonData.end())
+    {
+        fpsCapped = jsonData["fps_capped"];
+    }
+
+    if (!fpsCapped)
+    {
+        glfwSwapInterval(0);
+    }
 
     if (glewInit() != GLEW_OK)
     {
-        std::cerr << "glewInit() failure" << std::endl;
+        spdlog::error("glewInit() failure");
         return nullptr;
     }
 
     GLCall(glEnable(GL_BLEND));
     GLCall(glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA));
 
-    std::cout << "Using OpenGL version: " << glGetString(GL_VERSION) << std::endl;
+    std::string glVersion = reinterpret_cast<const char*>(glGetString(GL_VERSION));
```

Comment:
the input type here should be `const char *`, alternatively you could make this auto and save these little weird conversions


```
const char* glVersion = reinterpret_cast<const char*>(glGetString(GL_VERSION));

auto glVersion = reinterpret_cast<const char*>(glGetString(GL_VERSION));
```

Basically you're superfluously building a std::string here, which seems minorly excessing